StreamVideoTests/WebRTC/v2/WebRTCCoorindator_Tests.swift

```diff
@@ -0,0 +1,565 @@
+        await assertNilAsync(await subject.stateAdapter.publisher)
+        await assertNilAsync(await subject.stateAdapter.subscriber)
+        await assertNilAsync(await subject.stateAdapter.statsReporter)
+        await assertNilAsync(await subject.stateAdapter.sfuAdapter)
+        await assertEqualAsync(await subject.stateAdapter.token, "")
+        await assertEqualAsync(await subject.stateAdapter.sessionID, "")
+        await assertEqualAsync(await subject.stateAdapter.ownCapabilities, [])
+        await assertEqualAsync(await subject.stateAdapter.participants, [:])
+        await assertEqualAsync(await subject.stateAdapter.participantsCount, 0)
+        await assertEqualAsync(await subject.stateAdapter.anonymousCount, 0)
+        await assertEqualAsync(await subject.stateAdapter.participantPins, [])
+    }
+
+    // MARK: - leave
+
+    func test_leave_shouldTransitionStateMachineToLeaving() async throws {
+        /// We are going to try connecting and allow it to fail. We then stub the internet availability, in
+        /// order to transition to ``.disconnected`` and wait. In the meanwhile we are going to trigger
+        /// a manual `leave`.
+        let mockInternetConnection = MockInternetConnection()
+        InjectedValues[\.internetConnectionObserver] = mockInternetConnection
+        mockInternetConnection.subject.send(.unavailable)
+
+        try await subject
+            .connect(callSettings: nil, ring: true)
+        await wait(for: 1)
+
+        await assertTransitionToStage(
+            .leaving,
+            operation: { self.subject.leave() }
+        ) { _ in }
+    }
+
+    // MARK: - changeCameraMode
+
+    func test_changeCameraMode_shouldUpdateCameraPosition() async throws {
+        try await prepareAsConnected()
+        let mockPublisher = try await XCTAsyncUnwrap(
+            await subject
+                .stateAdapter
+                .publisher as? MockRTCPeerConnectionCoordinator
+        )
+
+        try await subject.changeCameraMode(position: .back)
+
+        await assertEqualAsync(
+            await subject.stateAdapter.callSettings.cameraPosition,
+            .back
+        )
+        XCTAssertEqual(
+            mockPublisher.recordedInputPayload(
+                AVCaptureDevice.Position.self,
+                for: .didUpdateCameraPosition
+            )?.first,
+            .back
+        )
+    }
+
+    // MARK: - changeAudioState
+
+    func test_changeAudioState_shouldUpdateAudioState() async throws {
+        await subject.changeAudioState(isEnabled: false)
+
+        await assertEqualAsync(
+            await subject.stateAdapter.callSettings.audioOn,
+            false
+        )
+    }
+
+    // MARK: - changeVideoState
+
+    func test_changeVideoState_shouldUpdateVideoState() async throws {
+        /// This can't be test due to the preprocessor macro we are using
+        /// in the CallSettings init.
+//        await subject.changeVideoState(isEnabled: false)
```

Comment:
Ah, I remember that :) Still would prefer to remove the test over having commented out code.